test/rbi/rewriters/flatten_scopes_test.rb

```diff
@@ -0,0 +1,186 @@
+# typed: true
+# frozen_string_literal: true
+
+require "test_helper"
+
+module RBI
+  class FlattenScopesTest < Minitest::Test
+    def test_flatten_scopes_with_empty_scopes
+      rbi = RBI::Parser.parse_string(<<~RBI)
+        module A
+          class B
+            class C
+              module D; end
+            end
+          end
+        end
+
+        module E; end
+      RBI
+
+      rbi.flatten_scopes!
+
+      assert_equal(<<~RBI, rbi.string)
+        module A; end
+        module E; end
+        class A::B; end
+        class A::B::C; end
+        module A::B::C::D; end
+      RBI
+    end
+
+    def test_flatten_scopes_with_nonempty_scopes
+      rbi = RBI::Parser.parse_string(<<~RBI)
+        module A
+          A1 = 42
+
+          class B
+            class C
+              C1 = 42
+
+              module D; end
+            end
+          end
+        end
+
+        module E
+          E1 = 42
+        end
+      RBI
+
+      rbi = RBI::Tree.new
+      scope1 = RBI::Module.new("A")
+      scope1 << RBI::Const.new("A1", "42")
+      scope2 = RBI::Class.new("B")
+      scope3 = RBI::Class.new("C")
+      scope3 << RBI::Const.new("C1", "42")
+      scope4 = RBI::Module.new("D")
+      scope5 = RBI::Module.new("E")
+      scope5 << RBI::Const.new("E1", "42")
+      scope3 << scope4
+      scope2 << scope3
+      scope1 << scope2
+      rbi << scope1
+      rbi << scope5
```

Comment:
I don't think we need this anymore

```suggestion
```

Reply:
Whoops, missed that. Just fixed-up that commit and force-pushed.